### PR TITLE
util: fix race condition in WaitForFile

### PR DESCRIPTION
**Pull request by giuseppe**
enable polling also when using inotify.  It is generally useful to
have it as under high load inotify can lose notifications.  It also
solves a race condition where the file is created while the watcher
is configured and it'd wait until the timeout and fail.

Closes: https://github.com/containers/libpod/issues/2942

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Is this for slirp4netns API socket? `slirp4netns --ready-fd` does not work?

**Comment by giuseppe** [comment]
> Is this for slirp4netns API socket? `slirp4netns --ready-fd` does not work?

IIRC, the ready-fd is written by the slirp4netns `child` while the api socket is managed by the parent so we cannot assume it is ready when ready-fd returned

**Comment by rhatdan** [comment]
LGTM
@mheon @vrothberg @baude @TomSweeneyRedHat PTAL

**Comment by giuseppe** [comment]
> I think we can simplify the code with using `case <- time.After(...)` in line 129. Then we could delete lines 108-122 entirely.

yes good point.  I'll change it

**Comment by AkihiroSuda** [comment]
> we cannot assume it is ready when ready-fd returned

I feel this is a bug. I opened https://github.com/rootless-containers/slirp4netns/issues/90

**Comment by giuseppe** [comment]
/hold cancel

Pushed a new version that simplifies it even further and doesn't use any go routine

File: libpod/util.go [hunk 1]
```diff
@@ -91,62 +91,39 @@ func MountExists(specMounts []spec.Mount, dest string) bool {
...
+
+	for {
+		select {
+		case <-time.After(25 * time.Millisecond):
+			// Check periodically for the file existence.  It is needed
+			// if the inotify watcher could not have been created.  It is
+			// also useful when using inotify as if for any reasons we missed
+			// a notification, we won't hang the process.
+			_, err := os.Stat(path)
+			if err == nil {
```
Comment (rhatdan) [inline libpod/util.go]: Should we be checking if os.IsNotExist(err) and blowing up if the err is something else?

Comment (giuseppe) [reply libpod/util.go]: I had moved the existing code, but yes that is better.  I've pushed a new version where it checks the error type.

**Comment by rhatdan** [comment]
LGTM, 
Lets merge when tests pass

**Comment by mheon** [comment]
Tests are very angry

**Comment by mheon** [comment]
Looks like exec broke

**Comment by giuseppe** [comment]
@mheon tests are passing now

**Pull request merged**